local-infrastructure/README.md

```diff
@@ -0,0 +1,126 @@
+
+Navigate to the Portal Base project, and then go to the Settings tab (bottom left of the sidebar). Then select the Authentication button, and then OAuth. As we are leveraging OIDC, select the OpenID tab. Here, you will want to use the well-known openid configuration to fill in these fields:
+
+| Field | Value |
+| --- | --- |
+| Authorize URI | well-known authorization_endpoint |
+| Token URI | well-known token_endpoint |
+| Client ID | Your Client ID |
+| Client Secret | Put in "something" - your client is public and doesn't need a secret |
+| Authorization Method | Body |
+| User Info (claims) URI | well-known userinfo_endpoint |
+
+Leave the Scope field blank, and ensure that under the Roles section, there is one entry whre Claim is empty, Value is empty, and Role is set to "Authenticated".
+
+#### Login Form
+
+Once the Portal Base project has the OAuth settings configured, you will need to manually create OAuth buttons in the Login form as mentioned in the [docs](https://help.form.io/integrations/oauth/#add-oauth-button-to-form).
+
+In the Portal Base project, select the Forms tab, and then edit the "User Login Form". Drag and drop in a new button under the existing Log In button. Set the Label to "IDIR Login", change Action to OAuth, and set OAuth Provider to OpenID. For optional aesthetics, we suggest setting the button theme to something other than Primary, and enabling the Block Button option. Then save the form.
+
+Once the button is set up, you need to set up an action for that button. Visit the actions tab, and then add a new "OAuth (Premium)" action. For Action Settings, set OAuth Provider to "OpenID", Action to "Login Existing Resource", Resource to "User", and Sign-in with OAuth Button to "IDIR Login". Then save this action.
+
+At this stage, you should be able to create a new incognito window, and attempt to login to Formio through IDIR/Keycloak. Upon completion, you should expect to see an error along the lines of "OpenID account has not yet been linked". The next section will resolve this as you need to register your existence with Formio first.
+
+#### Register Form
+
+In the Portal Base project, select the Forms tab, and then edit the "User Registration Form". Drag and drop in a new button under the existing Log In button. Set the Label to "IDIR Register", change Action to OAuth, and set OAuth Provider to OpenID. For optional aesthetics, we suggest setting the button theme to something other than Primary, and enabling the Block Button option.
+
+Once the button is set up, you need to set up an action for that button. Visit the actions tab, and then add a new "OAuth (Premium)" action. For Action Settings, set OAuth Provider to "OpenID", Action to "Register New Resource", Resource to "User", Role to "Authenticated" and Sign-in with OAuth Button to "IDIR Register".
+
+The Field Mappings section maps JWT values provided by Keycloak into formio. You will want to set the following map claims and then save the action.
```

Comment:
you should mention here that, the Fields you want to map to, wont show in the dropdown until you save the action and then go back in to edit it again

Reply:
Added